lib/Backends/OpenCL/OpenCL.h

```diff
@@ -134,28 +131,33 @@ class OpenCLFunction final : public CompiledFunction {
 private:
   /// Copy the value from a device to a provided buffer.
   /// \returns number of copied bytes.
-  uint64_t copyValueFromDevice(const Value *v, void *buf = nullptr);
+  uint64_t copyValueFromDevice(const Value *v,
+                               runtime::OpenCLDeviceBindings *devBindings,
+                               void *buf = nullptr);
   /// Copy value from the provided buffer to the device.
   /// \returns number of copied bytes.
-  uint64_t copyValueToDevice(const Value *v, void *buf = nullptr);
+  uint64_t copyValueToDevice(const Value *v,
+                             runtime::OpenCLDeviceBindings *devBindings,
+                             void *buf = nullptr);
   /// Fill the device \p buffer with a given \p value.
   /// \param len number of buffer elements to be filled by the \p value.
   /// Elements are considered to be of the type described by \p elemKind.
   void fillBuffer(cl_mem buffer, uint64_t start, uint64_t len, float value,
-                  ElemKind elemKind);
+                  ElemKind elemKind,
+                  runtime::OpenCLDeviceBindings *devBindings);
 
   /// Execution a convolution instruction which uses NCHW format.
   void executeNCHWConvolution(const ConvolutionInst *CC,
                               ExecutionContext *executionContext);
   /// Allocate a device buffer of required \p size.
-  cl_mem allocDeviceBuffer(uint64_t size);
+  cl_mem allocDeviceBuffer(uint64_t size, cl_context clContext);
   /// Frees a device buffer.
   void freeDeviceBuffer(cl_mem buf);
 
   /// Create kernel with a given \p name from a \p program.
   /// If \p program is nullptr, try to find the kernel with a given \p name
   /// in any of compiled programs.
-  cl_kernel createKernel(const std::string &name, cl_program program = nullptr);
+  cl_kernel createKernel(const std::string &name, cl_program program);
```

Comment:
Now that this can no longer be null, should we check its not in the implementation?